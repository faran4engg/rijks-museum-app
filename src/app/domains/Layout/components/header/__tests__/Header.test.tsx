import React from 'react';
import { act, create } from 'react-test-renderer';
import { createRenderer } from 'react-test-renderer/shallow';

import Header from '../Header';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

jest.mock('react-router-dom', () => ({
  NavLink: () => '<NavLink />',
}));

const renderer = createRenderer();

describe('<Header />', () => {
  const handleIsSidebarOpenMock = jest.fn();

  it('should render and match the snapshot', () => {
    renderer.render(<Header handleIsSidebarOpen={handleIsSidebarOpenMock} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  describe('toggleDark', () => {
    const renderComponent = () =>
      create(<Header handleIsSidebarOpen={handleIsSidebarOpenMock} />);

    it('should set dark mode when clicked', () => {
      Object.defineProperty(window, 'localStorage', {
        value: {
          getItem: jest.fn(() => 'light'),
          setItem: jest.fn(() => null),
        },
        writable: true,
      });
      const component = renderComponent();
      const root = component.root;
      const toggleEl = root.findAllByType('button')[1];
      act(() => {
        toggleEl.props.onClick();
      });
      expect(window.localStorage.setItem).toHaveBeenCalled();
      expect(window.localStorage.setItem).toHaveBeenCalledTimes(1);
      expect(window.localStorage.setItem).toHaveBeenCalledWith(
        'twTheme',
        'dark',
      );
      expect(component).toMatchSnapshot();
    });

    it('should set light mode when clicked', () => {
      Object.defineProperty(window, 'localStorage', {
        value: {
          getItem: jest.fn(() => 'dark'),
          setItem: jest.fn(() => null),
        },
        writable: true,
      });

      const component = renderComponent();
      const root = component.root;
      const toggleEl = root.findAllByType('button')[1];
      act(() => {
        toggleEl.props.onClick();
      });
      expect(window.localStorage.setItem).toHaveBeenCalled();
      expect(window.localStorage.setItem).toHaveBeenCalledTimes(1);
      expect(window.localStorage.setItem).toHaveBeenCalledWith(
        'twTheme',
        'light',
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('Menu Icon', () => {
    const renderComponent = () =>
      create(<Header handleIsSidebarOpen={handleIsSidebarOpenMock} />);

    it('should open side bar', () => {
      const component = renderComponent();
      const root = component.root;
      const MenuEl = root.findAllByType('button')[0];
      act(() => {
        MenuEl.props.onClick();
      });

      expect(handleIsSidebarOpenMock).toHaveBeenCalledTimes(1);
    });
  });
});
