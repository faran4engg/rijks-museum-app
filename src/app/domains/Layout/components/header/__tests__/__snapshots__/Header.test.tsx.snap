// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match the snapshot 1`] = `
<header
  className="z-40 py-2 shadow-md lg:shadow-none shadow-bottom bg-light-mode dark:bg-dark-mode"
>
  <div
    className="container flex items-center justify-between h-full px-3 mx-auto text-theme-color-light dark:text-theme-color-dark"
  >
    <button
      aria-label="Menu"
      className="p-1 -ml-1 rounded-md lg:hidden focus:outline-none focus:shadow-outline-green-200"
      onClick={[MockFunction]}
    >
      <SvgIcon
        className="w-6 h-6"
        icon="MenuIcon"
      />
    </button>
    <div
      className="flex justify-center flex-1"
    >
      <NavLink
        className="block lg:hidden"
        to="/home"
      >
        <SvgIcon
          className="w-28 h-10"
          icon="QardioLogoIcon"
        />
      </NavLink>
    </div>
    <ul
      className="flex items-center flex-shrink-0 space-x-6"
    >
      <li
        className="relative text-theme-color-light dark:text-theme-color-dark"
      >
        <button
          aria-haspopup="true"
          aria-label="Notifications"
          className="relative align-middle rounded-md focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="w-5 h-5 mx-auto animate-spin"
          >
            ðŸŒš
          </span>
          <span
            className="absolute top-0 right-0 inline-block w-3 h-3 transform translate-x-1 -translate-y-1 bg-red-600 border-2 border-white rounded-full animate-ping dark:border-gray-800"
          />
        </button>
      </li>
    </ul>
  </div>
</header>
`;

exports[`<Header /> toggleDark should set dark mode when clicked 1`] = `
<header
  className="z-40 py-2 shadow-md lg:shadow-none shadow-bottom bg-light-mode dark:bg-dark-mode"
>
  <div
    className="container flex items-center justify-between h-full px-3 mx-auto text-theme-color-light dark:text-theme-color-dark"
  >
    <button
      aria-label="Menu"
      className="p-1 -ml-1 rounded-md lg:hidden focus:outline-none focus:shadow-outline-green-200"
      onClick={[MockFunction]}
    >
      <svg
        aria-hidden="true"
        className="w-6 h-6"
      >
        menu.svg
      </svg>
    </button>
    <div
      className="flex justify-center flex-1"
    >
      &lt;NavLink /&gt;
    </div>
    <ul
      className="flex items-center flex-shrink-0 space-x-6"
    >
      <li
        className="relative text-theme-color-light dark:text-theme-color-dark"
      >
        <button
          aria-haspopup="true"
          aria-label="Notifications"
          className="relative align-middle rounded-md focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="w-5 h-5 mx-auto animate-spin"
          >
            ðŸ’¡
          </span>
          <span
            className="absolute top-0 right-0 inline-block w-3 h-3 transform translate-x-1 -translate-y-1 bg-red-600 border-2 border-white rounded-full animate-ping dark:border-gray-800"
          />
        </button>
      </li>
    </ul>
  </div>
</header>
`;

exports[`<Header /> toggleDark should set light mode when clicked 1`] = `
<header
  className="z-40 py-2 shadow-md lg:shadow-none shadow-bottom bg-light-mode dark:bg-dark-mode"
>
  <div
    className="container flex items-center justify-between h-full px-3 mx-auto text-theme-color-light dark:text-theme-color-dark"
  >
    <button
      aria-label="Menu"
      className="p-1 -ml-1 rounded-md lg:hidden focus:outline-none focus:shadow-outline-green-200"
      onClick={[MockFunction]}
    >
      <svg
        aria-hidden="true"
        className="w-6 h-6"
      >
        menu.svg
      </svg>
    </button>
    <div
      className="flex justify-center flex-1"
    >
      &lt;NavLink /&gt;
    </div>
    <ul
      className="flex items-center flex-shrink-0 space-x-6"
    >
      <li
        className="relative text-theme-color-light dark:text-theme-color-dark"
      >
        <button
          aria-haspopup="true"
          aria-label="Notifications"
          className="relative align-middle rounded-md focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="w-5 h-5 mx-auto animate-spin"
          >
            ðŸŒš
          </span>
          <span
            className="absolute top-0 right-0 inline-block w-3 h-3 transform translate-x-1 -translate-y-1 bg-red-600 border-2 border-white rounded-full animate-ping dark:border-gray-800"
          />
        </button>
      </li>
    </ul>
  </div>
</header>
`;
